### PR TITLE
Extract Hibernate and database related code to separate package

### PR DESCRIPTION
**Pull request by jmuia**

### CODE REVIEW

File: misk/hibernate/src/main/kotlin/misk/hibernate/SchemaMigrator.kt [hunk 1]
```diff
@@ -37,10 +36,7 @@ internal class SchemaMigrator(
 ) {
   /** Returns a map from version to path. */
   fun availableMigrations(): NavigableMap<Int, String> {
-    if (config.migrations_path == null) {
-      return TreeMap()
-    }
-    val resources = resourceLoader.list(config.migrations_path)
```
Comment (jmuia) [inline misk/hibernate/src/main/kotlin/misk/hibernate/SchemaMigrator.kt]: I've refactored this as smart casts do not work across packages (an alternative would be `val resources = resourceLoader.list(config.migrations_path!!`)

File: settings.gradle [hunk 2]
```diff
@@ -1,4 +1,6 @@
 include ':misk/testing'
```
Comment (jmuia) [inline settings.gradle]: I'd like to standardize this to colons over slashes.

**Comment by jmuia** [comment]
I tried my best to avoid redundant/unnecessary dependencies when moving them around. I'll find and apply a tool to do this automatically in a separate PR.

File: misk/hibernate/gradle.properties [hunk 3]
```diff
@@ -0,0 +1,4 @@
+POM_ARTIFACT_ID=misk-hibernate
+POM_NAME=Misk
+POM_DESCRIPTION=An open source application container from Square
```
Comment (swankjesse) [inline misk/hibernate/gradle.properties]: mind renaming to `misk-hibernate` ?

**Pull request merged**